include/SFML/System/String.hpp

```diff
@@ -39,6 +39,39 @@
 
 namespace sf
 {
+////////////////////////////////////////////////////////////
+/// \brief Character traits for std::uint8_t
+///
+////////////////////////////////////////////////////////////
+struct SFML_SYSTEM_API U8StringCharTraits
+{
+    ////////////////////////////////////////////////////////////
+    /// \brief Type traits and static members
+    ///
+    /// These type traits and static members meet the requirements
+    /// for custom character traits.
+    /// https://en.cppreference.com/w/cpp/string/char_traits
+    ///
+    ////////////////////////////////////////////////////////////
+    using char_type = std::uint8_t;
+
```

Comment:
You may not need them right now, but it's also not great if it breaks down the road because you didn't add all the required functionality, especially since it's trivial to add right now. A string implementation is free to use all of it.

Reply:
If you think it's trivial then would you mind adding the remaining missing functions?